Lib/pathlib.py

```diff
@@ -862,7 +788,7 @@ def samefile(self, other_path):
             other_st = other_path.stat()
         except AttributeError:
             other_st = self.__class__(other_path).stat()
-        return os.path.samestat(st, other_st)
+        return self._flavour.samestat(st, other_st)
```

Comment:
Basing `samefile()` on `stat()` is problematic in Windows. The volume serial number is used for `st_dev`. The VSN of a filesystem can be any 32-bit value. It isn't forced to be unique, but usually it's a random value that's likely to be unique. The file ID is used for `st_ino`. It's 0 if a filesystem doesn't support file IDs. WebDAV is a common example that always returns 0 for both the VSN and file IDs, so the `stat()` result is completely worthless for `samefile()` within or between WebDAV filesystems. This isn't buggy behavior. The WebDAV filesystem in Windows is completely within its rights to behave this way according to documented protocols.

This method should call `self._flavour.samefile()` instead of `samestat()`. It currently has the same design flaw, but if we're going to fix this anywhere, it's by moving the Windows implementations of `samefile()` and `sameopenfile()` back into `ntpath`. There are open issues for this, but there's no momentum pushing for a resolution.

Reply:
I agree, but I don't think this is the right place to fix it. Everything I add to this PR makes it harder to review and less likely to land.

Reply:
When implementing `samefile()`, I should think it would be an easy sell to use `self._flavour.samefile()`. This is more flexible for subclasses since it doesn't assume any `stat()` semantics that may not be supported. The fact that `st_dev` and `st_ino` are unreliable in Windows is just one example.